src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -259,4 +260,179 @@ public void systemEndsPatientVisit() {
+		allergyPage.waitForPage();
+	}
+
+	@Then("the system adds a new Allergy")
+	public void systemAddsAllergy() {
+		assertTrue(textExists("Codeine"));
+	}
+
+	// User story: Add New Condition
+	@When("a user clicks on Conditions on Patient dashboard")
+	public void loadManageConditions() {
+		conditionsPage = (ConditionsPage) dashboardPage
+				.clickOnConditionsWidgetLink().waitForPage();
+	}
+
+	@Then("system loads Conditions Page")
+	public void loadsManageConditionsPage() {
+		assertTrue(conditionsPage.containsText("Conditions"));
+	}
+
+	@When("user clicks on Add new condition button")
+	public void clicksAddNewCondition() {
+		conditionPage = (ConditionPage) conditionsPage.clickOnAddNewCondition()
+				.waitForPage();
+	}
+
+	@And("user enters patient condition")
+	public void userEntersExistingCondition() {
+		conditionPage.typeInCondition(CONDITION);
+	}
+
+	@And("user clicks on save button")
+	public void clicksSaveCondition() {
+		conditionPage.clickSave();
+	}
+
+	@Then("system adds New Condition in Conditions table")
+	public void systemAddsNewCondition() {
+		assertNotNull(conditionsPage.getConditionsList());
+		dashboardPage = conditionsPage.clickReturn();
+	}
+
+	// Scenario: Patient Dashboard adding Allergy2
+	// User story: Adding and deleting New Allergy2
+	@When("a user clicks on Allergy2 link from Patient dashboard page2")
+	public void loadManageAllergy2() {
+		allergyPage = (AllergyPage) dashboardPage.clickOnAllergiesWidgetLink()
+				.waitForPage();
+	}
+
+	@Then("the system loads Allergy2 board page")
+	public void loadsManageAllergy2() {
+		assertEquals(getElement(patientHeaderId).getText(),
+				getElement(patientHeaderId).getText());
+		assertTrue(textExists("Allergies"));
+	}
+
+	@When("a user clicks Add New Allergy2 button")
+	public void addNewAllergy2() {
+		allergyPage.clickOnAddNewAllergy();
+	}
+
+	@Then("the system loads Add New Allergy2 dashboard")
+	public void loadsAddNewAllergy2() {
+		assertTrue(textExists("Add New Allergy"));
+	}
+
+	@And("the user selects a new Allergy2")
+	public void selectNewAllergy2() {
+		allergyPage.clickOn(By.id("allergen-162297-label"));// Penicillins
+	}
+
+	@And("the user clicks on the Save button2")
+	public void saveAllergy2() {
+		allergyPage.clickOn(By.id("addAllergyBtn"));
+		allergyPage.waitForPage();
+	}
+
+	@Then("the system adds a new Allergy2")
+	public void systemAddsAllergy2() {
+		assertTrue(textExists("Penicillins"));
+	}
+
+	@When("the user clicks on the delete button from the patient dashboard")
+	public void deleteAllergy2() {
+		allergyPage.clickOnDeleteAllergy();
+	}
+
+	@And("the system loads Remove Allergy dashboard")
+	public void confirmDeleteAllergy2() {
+		allergyPage.clickOnConfirmDeleteAllergy();
+	}
+
+	@Then("system displays no allergy in the Allergies table")
+	public void systemRemovesAllergy2() {
+		assertTrue(textExists("Unknown"));
+	}
+
+	// Scenario: Patient Dashboard adding Condition2
+	// User story: Adding and deleting New Condition2
```

Comment:
Refer to my comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/137/files#r717582844)